ground/src/main/java/com/google/android/ground/ui/home/mapcontainer/LoiCardAdapter.kt

```diff
@@ -44,63 +45,58 @@ class LoiCardAdapter : RecyclerView.Adapter<ViewHolder>() {
 
   /** Binds [LocationOfInterest] data to [ViewHolder]. */
   override fun onBindViewHolder(holder: ViewHolder, position: Int) {
-    holder.bind(itemsList[position])
+    val loi: LocationOfInterest = itemsList[position]
+
+    holder.bind(loi)
 
     // Add highlight border if selected.
     holder.binding.wrapperView.background =
       ResourcesCompat.getDrawable(
         holder.itemView.context.resources,
-        if (selectedIndex == position) {
+        if (focusedIndex == position) {
           R.drawable.border
         } else {
           R.color.colorBackground
         },
         null
       )
 
-    // Handle action buttons.
-    holder.binding.loiCard.setOnClickListener { handleItemClicked(position) }
-    holder.binding.start.setOnClickListener { handleButtonClicked(position) }
-    holder.binding.review.setOnClickListener { TODO() }
-    holder.binding.markComplete.setOnClickListener { TODO() }
+    val isCaptionMissing = loi.caption.isNullOrEmpty()
+    holder.binding.loiIcon.visibility = if (isCaptionMissing) View.GONE else View.VISIBLE
+    holder.binding.loiName.visibility = if (isCaptionMissing) View.GONE else View.VISIBLE
+    holder.binding.loiCard.setOnClickListener { collectDataCallback.invoke(loi) }
```

Comment:
Is it possible to use data bindings for this instead of setting programmatically? If it's difficult or not possible please disregard.

Reply:
We can do it by passing a boolean variable "isCaptionMissing" to the xml. I tried directly using `.isNullOrEmpty()` in data binding but it didn't work.

Reply:
It seems data binding still uses Java, so extension functions like `.isNullOrEmpty()` would need to be imported and called statically ([example](https://stackoverflow.com/a/59756207/9986466)). Ideally we'd model view states in the `ViewModel` class rather than expect the layout to compute them; in this case the states in the VM should already be non-null and reflect any defaults or overrides. This way we can unit test them more easily, and we have a singular place to look for view logic.

Reply:
Thanks for the tip! Done